src/analysis/normed_space/basic.lean

```diff
@@ -359,35 +359,41 @@ lemma add_monoid_hom.continuous_of_bound (f : α →+ β) (C : ℝ) (h : ∀x, 
 
 section nnnorm
 
-/-- Version of the norm taking values in nonnegative reals. -/
-def nnnorm (a : α) : ℝ≥0 := ⟨norm a, norm_nonneg a⟩
+/-- Auxiliary class, endowing a type `α` with a function `nnnorm : α → ℝ≥0`. -/
+class has_nnnorm (α : Type*) := (nnnorm : α → ℝ≥0)
 
-@[simp, norm_cast] lemma coe_nnnorm (a : α) : (nnnorm a : ℝ) = norm a := rfl
+export has_nnnorm (nnnorm)
 
-lemma nndist_eq_nnnorm (a b : α) : nndist a b = nnnorm (a - b) := nnreal.eq $ dist_eq_norm _ _
+notation `∥₊`:1024 e:1 `∥`:1 := nnnorm e
```

Comment:
If someone has a better idea for the notation, pleas don't hesitate to gives suggestion. The more natural ``∥`:1024 e:1 `∥₊`:1` doesn't work, I think because Lean gets confused with `∥ ∥`.

Reply:
Perhaps `┃ x ┃`? That's unicode character `box drawings heavy vertical`.

Reply:
This seems pretty complicated to type, doesn't it? For example in VS Code I don't know how to write it.

Reply:
Hmm, in LTE I successfully used `∥x∥₊` for `Mbar`. I would prefer that version over having the `∥₊x∥`. Maybe ping some notation experts on Zulip?